common/client-core/src/client/base_client/mod.rs

```diff
@@ -865,7 +873,17 @@ where
     fn construct_nym_api_client(
         config: &Config,
         user_agent: Option<UserAgent>,
+        custom_client: Option<nym_http_api_client::Client>,
```

Comment:
rather than providing custom client, I reckon it would be better to provide custom configuration to the client (which could be defined on the normal `Config` struct)